tasks/copy-files.yml

```diff
@@ -0,0 +1,22 @@
+---
+- name: "Ensure directory"
+  ansible.builtin.file:
+    path: "{{ item.dest | dirname }}"
+    state: directory
+    owner: "{{ app_user }}"
+    mode: "0700"
```

Comment:
You probably want to control the group for this and the loop below with `app_group`.